### PR TITLE
[JENKINS-50068] Add integration test for AffectedFilesResolver.

### PR DESCRIPTION
**Pull request by dmardin**
This Pull Request needs the Pull Request https://github.com/jenkinsci/analysis-core-plugin/pull/125 because the IntegrationTest is updated by a whitelist which excludes some files (.zip, .gz, .tar) from being renamed to "-issues.txt".

### CODE REVIEW

File: src/test/java/io/jenkins/plugins/analysis/warnings/AffectedFilesResolverIT.java [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+        String secondPartOfLink = getSourceLink(result);
+        HtmlPage contentPage = webClient.getPage(result.getOwner(),
+                DEFAULT_ENTRY_PATH + firstPartOfLink + secondPartOfLink);
+        DomElement domElement = contentPage.getElementById("main-panel");
+        DomNodeList<HtmlElement> list = domElement.getElementsByTagName("a");
+        removeElementsByTagName(list);
+
+        String content = convertFileToString(new File(j.jenkins.getWorkspaceFor(project) + "/" + EXTRACTED_FILE));
+
+        assertThat(contentPage.getElementById("main-panel").asText().replaceAll(WHITESPACE, "")).contains(
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/warnings/AffectedFilesResolverIT.java]: Funktioniert nicht auch `isEqualToIgnoringWhitespace` statt `contains`?

File: src/test/java/io/jenkins/plugins/analysis/warnings/AffectedFilesResolverIT.java [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+    /**
+     * Verifies that the AffectedFilesResolver produces an I/O error, when the affected files cannot copied.
+     */
+    @Test
+    public void shouldGetIOErrorBySearchingForAffectedFiles() {
+        FreeStyleProject project = createJobWithWorkspaceFile(ECLIPSE_TXT, ZIP_FILE);
+        enableWarnings(project);
+
+        String pathToExtractedFile = j.jenkins.getWorkspaceFor(project) + "/" + EXTRACTED_FILE;
+        if (System.getProperty("os.name").contains("Windows")) {
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/warnings/AffectedFilesResolverIT.java]: Das Setzen auf ReadOnly wäre in einer Methode schön, dann kann das überall verwendet werden

**Comment by dmardin** [comment]
Dieser PR braucht den FIX von: https://github.com/jenkinsci/analysis-core-plugin/pull/124

**Comment by uhafner** [comment]
Merged into https://github.com/jenkinsci/warnings-plugin/tree/3.0/src/test/java/io/jenkins/plugins/analysis/warnings/recorder

**Pull request closed without merging**